Lib/dataclasses.py

```diff
@@ -0,0 +1,776 @@
+                 'default_factory',
+                 'repr',
+                 'hash',
+                 'init',
+                 'compare',
+                 'metadata',
+                 '_field_type',  # Private: not to be used by user code.
+                 )
+
+    def __init__(self, default, default_factory, init, repr, hash, compare,
+                 metadata):
+        self.name = None
+        self.type = None
+        self.default = default
+        self.default_factory = default_factory
+        self.init = init
+        self.repr = repr
+        self.hash = hash
+        self.compare = compare
+        self.metadata = (_EMPTY_METADATA
+                         if metadata is None or len(metadata) == 0 else
+                         types.MappingProxyType(metadata))
+        self._field_type = None
+
+    def __repr__(self):
+        return ('Field('
+                f'name={self.name!r},'
+                f'type={self.type},'
+                f'default={self.default},'
+                f'default_factory={self.default_factory},'
+                f'init={self.init},'
+                f'repr={self.repr},'
+                f'hash={self.hash},'
+                f'compare={self.compare},'
+                f'metadata={self.metadata}'
+                ')')
+
+
+# This function is used instead of exposing Field creation directly,
+#  so that a type checker can be told (via overloads) that this is a
+#  function whose type depends on its parameters.
+def field(*, default=_MISSING, default_factory=_MISSING, init=True, repr=True,
+          hash=None, compare=True, metadata=None):
+    """Return an object to identify dataclass fields.
+
+    default is the default value of the field. default_factory is a
+    0-argument function called to initialize a field's value. If init
+    is True, the field will be a parameter to the class's __init__()
+    function. If repr is True, the field will be included in the
+    object's repr(). If hash is True, the field will be included in
+    the object's hash(). If compare is True, the field will be used in
+    comparison functions. metadata, if specified, must be a mapping
+    which is stored but not otherwise examined by dataclass.
+
+    It is an error to specify both default and default_factory.
+    """
+
+    if default is not _MISSING and default_factory is not _MISSING:
+        raise ValueError('cannot specify both default and default_factory')
+    return Field(default, default_factory, init, repr, hash, compare,
+                 metadata)
+
+
+def _tuple_str(obj_name, fields):
+    # Return a string representing each field of obj_name as a tuple
+    #  member. So, if fields is ['x', 'y'] and obj_name is "self",
+    #  return "(self.x,self.y)".
+
+    # Special case for the 0-tuple.
+    if len(fields) == 0:
+        return '()'
+    # Note the trailing comma, needed if this turns out to be a 1-tuple.
+    return f'({",".join([f"{obj_name}.{f.name}" for f in fields])},)'
```

Comment:
This would look clearer if extract the internal expression as a variable. With space after comma the result would conform PEP 8. The trailing comma is needed only for a 1-tuple.

Reply:
I'm not so concerned about this. It's generated code, and I don't particularly care how the generated code looks, as long as it's correct.